### ISSUE TITLE
Unable to filter using metadata

### ISSUE THREAD

**Issue by victoryhb** [lum-ai/odinson#328]
I am experimenting with a custom annotation pipeline to generate Odinson documents for indexing. After some trial and error (since I couldn't find related info in the docs:), I was able to index the documents along with some metadata. The current metadata of one of my files as shown via Odinson's REST API interface (/api/metadata/by-document-id, version 0.5.1 Snapshot) is something like the following:
```
[
  {
    "$type": "ai.lum.odinson.StringField",
    "name": "file",
    "string": "AAT"
  },
  {
    "$type": "ai.lum.odinson.StringField",
    "name": "corpus",
    "string": "BNC"
  }
]
```

However, when trying to filter documents with the filter: corpus=='BNC',  either using the parentQuery of /api/execute/pattern or defining the metadataFilters parameter through /api/execute/grammar,  no results were returned (the results were correctly returned without the filter). Below is the grammar I tested with:

```
metadataFilters: 
 - corpus=='BNC'

rules:
 - name: "example"
   label: GrammaticalSubject
   type: event
   pattern: |
     trigger = [lemma=have]
     subject  = >nsubj []
```

I wonder if I am doing anything wrong here?

**Comment by BeckySharp** [comment]
@victoryhb thanks, I'll work on reproducing this tonight.  BTW, there should be a PR very soon that will make it easier to add metadata (I thought I had merged it tbh, but i don't see it, which means I must have forgot to finish some last step...)

**Comment by BeckySharp** [comment]
> @victoryhb thanks, I'll work on reproducing this tonight. BTW, there should be a PR very soon that will make it easier to add metadata (I thought I had merged it tbh, but i don't see it, which means I must have forgot to finish some last step...)

ha! i'm an idiot, it _was_ merged in #319 though you're right that there isn't documentation about it...

**Comment by BeckySharp** [comment]
@victoryhb reproduced.  However, it's not a bug per se as much as poor documentation.
The StringField is not supported for metadata.
If you convert your metadata to this:
```
[
  {
    "$type": "ai.lum.odinson.TokensField",
    "name": "file",
    "tokens": ["AAT"]
  },
  {
    "$type": "ai.lum.odinson.TokensFIeld",
    "name": "corpus",
    "tokens": ["BNC"]
  }
]
```

it should work.  I will:

1. add a test for this advice
2. add an exception for StringFields as metadata
3. add info to the documentation

**Comment by victoryhb** [comment]
@BeckySharp Just tested again and problem solved! Thanks a lot! I wasn't expecting the need to use TokensField since I deduced from the example given by /api/metadata/by-document-id that the format should be something like the following:

```
[
  {
    "$type": "string",
    "name": "pmc",
    "string": "6360082",
    "store": true
  },
  {
    "$type": "string",
    "name": "publicationDate",
    "date": "2019-01-31",
    "store": true
  }
]
```
Great to hear you have made it easier to add metadata. Looking forward to seeing some example usage in the docs (or some pointers to example tests in the code). Thanks!

**Comment by BeckySharp** [comment]
> @BeckySharp Just tested again and problem solved! Thanks a lot! I wasn't expecting the need to use TokensField since I deduced from the example given by /api/metadata/by-document-id that the format should be something like the following:
> 
> ```
> [
>   {
>     "$type": "string",
>     "name": "pmc",
>     "string": "6360082",
>     "store": true
>   },
>   {
>     "$type": "string",
>     "name": "publicationDate",
>     "date": "2019-01-31",
>     "store": true
>   }
> ]
> ```
> 
> Great to hear you have made it easier to add metadata. Looking forward to seeing some example usage in the docs (or some pointers to example tests in the code). Thanks!

And you were right to deduce this, what happened is that the metadata query language got a total re-write in #305 , and I think that not everything was fully updated.  @myedibleenso @danebell does something in the `backend` need to change here?

**Comment by kwalcock** [comment]
> > @victoryhb thanks, I'll work on reproducing this tonight. BTW, there should be a PR very soon that will make it easier to add metadata (I thought I had merged it tbh, but i don't see it, which means I must have forgot to finish some last step...)
> 
> ha! i'm an idiot, it _was_ merged in #319 though you're right that there isn't documentation about it...

Be careful there, @BeckySharp.  Code of conduct also applies to self references.  Be nice to you.  :-)
